### PR TITLE
Enable differenct currencies

### PR DESCRIPTION
**Pull request by OmarHatem28**

### CODE REVIEW

**Comment by OmarHatem28** [comment]
@mkyq I noticed that there is already a widget `Picker` that has some of the desired UI while looking into CW-104 and 106, anyway it would have needed some refactoring to allow the required change currency design.
However, I am refactoring `Picker` right now while working on the new design and the new `CurrencyAlertDialog` in this PR will be removed then.

File: lib/view_model/buy/buy_amount_view_model.dart [hunk 1]
```diff
@@ -11,7 +11,8 @@ abstract class BuyAmountViewModelBase with Store {
   @observable
   String amount;
 
-  FiatCurrency get fiatCurrency => FiatCurrency.usd;
+  @observable
+  FiatCurrency fiatCurrency = FiatCurrency.usd;
```
Comment (mkyq) [inline lib/view_model/buy/buy_amount_view_model.dart:15]: set value inside of constructor

File: lib/src/widgets/currency_alert_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        child: Container(
+          color: Colors.transparent,
+          child: Container(
+            color: Color(0xff010615).withOpacity(0.8),
+            child: Center(
+              child: Column(
+                children: [
+                  Expanded(child: const SizedBox()),
+                  Text(
+                    "Change Currency",
```
Comment (mkyq) [inline lib/src/widgets/currency_alert_dialog.dart:26]: Add to localization file

File: lib/src/widgets/currency_alert_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.transparent,
+      body: GestureDetector(
+        onTap: () => Navigator.of(context).pop(),
+        child: Container(
+          color: Colors.transparent,
+          child: Container(
+            color: Color(0xff010615).withOpacity(0.8),
```
Comment (mkyq) [inline lib/src/widgets/currency_alert_dialog.dart:20]: Please move all unnamed colors into theme

**Review by mkyq — CHANGES_REQUESTED** [review]

**Pull request closed without merging**